dll/win32/shell32/droptargets/CFSDropTarget.cpp

```diff
@@ -72,7 +72,7 @@ HRESULT CFSDropTarget::_CopyItems(IShellFolder * pSFFrom, UINT cidl,
 
     TRACE ("(%p)->(%p,%u,%p)\n", this, pSFFrom, cidl, apidl);
 
-    STRRET strretFrom;
+    STRRET strretFrom = { 0 }; // Initialize strretFrom.pOleStr to NULL
     hr = pSFFrom->GetDisplayNameOf(NULL, SHGDN_FORPARSING, &strretFrom);
```

Comment:
https://learn.microsoft.com/en-us/windows/win32/api/shobjidl_core/nf-shobjidl_core-ishellfolder-getdisplaynameof
Can you re-check this method's SAL?

Reply:
It's an interface function: https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/include/psdk/shobjidl.idl;hb=0616df11e6a4c4fb29d0a44bc4d278608472b485#l271

Reply:
strret has no `[in]` attribute. initialization is callee responsibility.

Reply:
However that responsibility reach the failure cases?

Reply:
Normally, it's as follows: If the function fails, the `strretFrom` is left in an unknown state (that we don't have to care since we return on failure just below). However, if the function succeeds, the `strretFrom` should be correctly initialized (on output, of course).

Reply:
> When this method returns, contains a pointer to a STRRET structure in which to return the display name.

Reply:
That's (expressed in different words) what I've just said above:
> However, if the function succeeds, the strretFrom should be correctly initialized (on output, of course).

Reply:
@HBelusca, The function succeeds, but does not set the variable.
Here is my in-progress patch that you can run for yourself and possibly investigate further:
[CFSDropTarget_cpp_Diag.patch](https://github.com/reactos/reactos/files/12061631/CFSDropTarget_cpp_Diag.patch)
And here is my typical log file:
[copy-folder-11.txt](https://github.com/reactos/reactos/files/12061632/copy-folder-11.txt)

I believe that it is only after the BuildPathsList is called that this becomes a problem. Thanks.

Reply:
This looks like a bug from whatever module implements the `GetDisplayNameOf` function, whose implementation returns success without doing anything to the variable.